tests/console/golang.pm

```diff
@@ -0,0 +1,51 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test the 2 newest go versions by compiling and running man_or_boy.go
+# Maintainer: Dominik Heidler <dominik@heidler.eu>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use registration;
+
+sub mob_test {
+    assert_script_run "go build /home/$username/data/man_or_boy.go";
+    assert_script_run('test $(./man_or_boy) == "-67"');
+    assert_script_run 'rm man_or_boy';
+    assert_script_run("test \$(go run /home/$username/data/man_or_boy.go) == \"-67\"");
+}
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    if (is_sle() && !main_common::is_updates_tests()) {
+        add_suseconnect_product('sle-module-desktop-applications');
+        add_suseconnect_product(get_addon_fullname('sdk'));
+    }
+
+    script_run "zypper se go | grep ' go[0-9][0-9.]* '";
+    my $older_go = script_output "zypper se go | grep ' go[0-9][0-9.]* ' | awk -F '|' '{print \$2}' | tr -d ' ' | sort --version-sort | tail -2 | head -1";
+    my $latest_go = script_output "zypper se go | grep ' go[0-9][0-9.]* ' | awk -F '|' '{print \$2}' | tr -d ' ' | sort --version-sort | tail -1 | head -1";
```

Comment:
why three same calls of zypper? + more pipes  = better shell ? `grep` isn't needed with `awk`.

still `my output = script_ouput "zypper se go"` and then use perl to get versions will be much better

Reply:
I don't know how to grep with awk - when I want to grep I simply use grep.
I don't see any advantage on doing that stuff in perl - it would be much more complicated and unreadable (because its perl) and also I don't see a nice way to compare version numbers in perl without extra dependencies.
I guess zypper is fast enough so that it can be called three times. I could write the output of zypper to a file and cat it for the calls but I'm not sure if that would be an improvement.